src/components/molecules/Redirect/Redirect.tsx

```diff
@@ -0,0 +1,19 @@
+'use client';
+
+import { useSearchParams, redirect, RedirectType } from 'next/navigation';
+import path from 'path';
+
+export const Redirect = () => {
+  const searchParams = useSearchParams();
+
+  const requestDirectPath = searchParams.get('redirect');
```

Comment:
Can we assert that it starts with a `/` to explicitly prevent attempts to redirect outside of the domain.

Reply:
Sorry, I left a comment to this. But it was left in pending!
--

next.config.js rewrites prevents being sent outside of the domain.

We rewrite /redirect/<redirectPath> to <domain>/<redirectPath>

If you try http://localhost:3000/auth?redirect=https%3A%2F%2Fbadactor.com you'll be redirected to http://localhost:3000/https://badactor.com